creator-node/src/services/PrometheusMetric.js

```diff
@@ -0,0 +1,143 @@
+const client = require('prom-client')
+
+const PrometheusType = Object.freeze({
+  HISTOGRAM: 'histogram',
+  GAUGE: 'gauge'
+})
+
+/**
+ * Class to maintain library-specific logic
+ * for Prometheus-bound metrics
+ */
+class PrometheusMetric {
+  // use these static vars to avoid name collisions
+  // and register collectors to be run by the PrometheusMetricExporter
+  static histograms = {}
+  static gauges = {}
+  static registeredCollectors = {}
+
+  /**
+   * Helper to avoid naming collisions.
+   */
+  _initMetric(
+    name,
+    description,
+    labelNames,
+    collection,
+    PrometheusMetricClass
+  ) {
+    if (!(name in collection)) {
+      collection[name] = new PrometheusMetricClass({
+        name: name,
+        help: description,
+        labelNames: labelNames
+      })
+    }
+    this.metric = collection[name]
+  }
+
+  constructor(name, description, labelNames, metric_type) {
+    // set this.startTime to now()
+    this.resetTimer()
+
+    // set metric prefix of audius_project_
+    name = `audius_cn_${name}`
+
+    this.metricType = metric_type
```

Comment:
actually even better - lets not allow a default and force each caller to explicitly specify the metric type

Reply:
I moved the constructor to use named parameters, so I believe I've lost the ability to set default parameters, right?

For almost all cases, we'll want histograms since those "time things", but sometimes we'll want gauges to see what the current stat is.